src/codegen.c

```diff
@@ -0,0 +1,591 @@
+  
+  fprintf(f, "/* Define the settings structure */\n");
+  fprintf(f, "OSQPSettings %ssettings = {\n", prefix);
+  fprintf(f, "  0,\n"); // device
+  fprintf(f, "  OSQP_LINSYS_SOLVER,\n");
+  fprintf(f, "  0,\n"); // verbose
+  fprintf(f, "  %d,\n", settings->warm_starting);
+  fprintf(f, "  %d,\n", settings->scaling);
+  fprintf(f, "  0,\n"); // polishing
+  fprintf(f, "  (c_float)%.20f,\n", settings->rho);
+  fprintf(f, "  %d,\n", settings->rho_is_vec);
+  fprintf(f, "  (c_float)%.20f,\n", settings->sigma);
+  fprintf(f, "  (c_float)%.20f,\n", settings->alpha);
+  fprintf(f, "  %d,\n", settings->cg_max_iter);
+  fprintf(f, "  %d,\n", settings->cg_tol_reduction);
+  fprintf(f, "  (c_float)%.20f,\n", settings->cg_tol_fraction);
+  fprintf(f, "  %d,\n", settings->adaptive_rho);
+  fprintf(f, "  %d,\n", settings->adaptive_rho_interval);
+  fprintf(f, "  (c_float)%.20f,\n", settings->adaptive_rho_fraction);
+  fprintf(f, "  (c_float)%.20f,\n", settings->adaptive_rho_tolerance);
+  fprintf(f, "  %d,\n", settings->max_iter);
+  fprintf(f, "  (c_float)%.20f,\n", settings->eps_abs);
+  fprintf(f, "  (c_float)%.20f,\n", settings->eps_rel);
+  fprintf(f, "  (c_float)%.20f,\n", settings->eps_prim_inf);
+  fprintf(f, "  (c_float)%.20f,\n", settings->eps_dual_inf);
+  fprintf(f, "  %d,\n", settings->scaled_termination);
+  fprintf(f, "  %d,\n", settings->check_termination);
+  fprintf(f, "  (c_float)%.20f,\n", settings->time_limit);
+  fprintf(f, "  (c_float)%.20f,\n", settings->delta);
+  fprintf(f, "  %d,\n", settings->polish_refine_iter);
+  fprintf(f, "};\n\n");
+}
+
+
+/******
+* Info
+*******/
+
+static void write_info(FILE           *f,
+                       const OSQPInfo *info,
+                       const char     *prefix){
+  
+  fprintf(f, "/* Define the info structure */\n");
+  fprintf(f, "OSQPInfo %sinfo = {\n", prefix);
+  fprintf(f, "  \"%s\",\n", info->status);
+  fprintf(f, "  %d,\n", info->status_val);
+  fprintf(f, "  0,\n"); // status_polish
+  fprintf(f, "  (c_float)%.20f,\n", info->obj_val);
+  fprintf(f, "  (c_float)%.20f,\n", info->prim_res);
+  fprintf(f, "  (c_float)%.20f,\n", info->dual_res);
+  fprintf(f, "  %d,\n", info->iter);
+  fprintf(f, "  %d,\n", info->rho_updates);
+  fprintf(f, "  (c_float)%.20f,\n", info->rho_estimate);
+  fprintf(f, "  (c_float)0.0,\n"); // setup_time
+  fprintf(f, "  (c_float)0.0,\n"); // solve_time
+  fprintf(f, "  (c_float)0.0,\n"); // update_time
+  fprintf(f, "  (c_float)0.0,\n"); // polish_time
+  fprintf(f, "  (c_float)0.0,\n"); // run_time
+  fprintf(f, "};\n\n");
+}
+
+
+/**********
+* Solution
+***********/
+
+static void write_solution(FILE       *f,
+                           c_int       n,
+                           c_int       m,
+                           const char *prefix){
+
+  fprintf(f, "/* Define the solution structure */\n");
+  fprintf(f, "c_float %ssol_x[%d];\n", prefix, n);
+  if (m > 0) fprintf(f, "c_float %ssol_y[%d];\n", prefix, m);
```

Comment:
I think an underscore should also follow the prefix on these internal variables as well.

Reply:
After discussion, it will be expected the user provide an underscore in their prefix if they want to have underscores separating the prefix from the variable name.